aten/src/THC/THCReduce.cuh

```diff
@@ -63,58 +61,67 @@ kernelReduceNoncontigDim_shared(TensorInfo<T, IndexType> out,
     //for(IndexType i=threadIdx.y; i<reductionSize; i+=blockDim.y){
     //  local_reg += in[inOffset + i * reductionStride];
     //}
-    for(IndexType i=threadIdx.y; i<reductionSize; i+=blockDim.y*4){
-      if(i + blockDim.y * 3 < reductionSize){
-        load_reg[0] = modifyOp(in.data[inOffset + (i + blockDim.y * 0) * reductionStride]);
-        load_reg[1] = modifyOp(in.data[inOffset + (i + blockDim.y * 1) * reductionStride]);
-        load_reg[2] = modifyOp(in.data[inOffset + (i + blockDim.y * 2) * reductionStride]);
-        load_reg[3] = modifyOp(in.data[inOffset + (i + blockDim.y * 3) * reductionStride]);
+    for(IndexType i=threadIdx.y; i<reductionSize; i+=blockDim.y*4) {
+      if (i + blockDim.y * 3 < reductionSize) {
+        const AccT val0 = ScalarConvert<T, AccT>::to(in.data[inOffset + i * reductionStride]);
```

Comment:
can you please use `scalar_cast`? It's much more readable

Reply:
Happily.